tests/test_methods_nai.py

```diff
@@ -0,0 +1,58 @@
+"""This test aims to compare to previous PDA and PDAW results that were published and compared to exact QD.
```

Comment:
Could you tell me which version of the code specifically (commit hash ideally) did you use to generate the reference data?

Reply:
I used the last version on main (commit [629117a](https://github.com/PHOTOX/promdens/commit/629117adc7b0b5d24627375cb221ccea13b074df)) to generate the reference data. I created the data today and compared with QD.

Reply:
Okay cool. Would be also good to verify that you get the same numbers with the older versions of the code before I started to mess with it. :smile:

Reply:
I was doing the same test before we put it on GitHub and it passed. I'm not comparing the numbers but I always compared with QD and also tested that it wouldn't work if you mess up with the results. 
Just to make sure, I just tried two other pulses published (Gaussian and Lorentzian) and I still get the same as QD (one results attached). Thus, I believe this version is stable and working.

<img width="389" alt="Screenshot 2024-09-13 at 18 30 36" src="https://github.com/user-attachments/assets/d7f786ca-21e6-4574-98d3-1274b3e2f010">